core/pkg/resultshandling/printer/v2/resourcetable_test.go

```diff
@@ -0,0 +1,485 @@
+		controls       []resourcesresults.ResourceAssociatedControl
+		expectedLen    int
+	}{
+		{
+			name:           "Empty controls",
+			summaryDetails: reportsummary.SummaryDetails{},
+			controls:       []resourcesresults.ResourceAssociatedControl{},
+			expectedLen:    0,
+		},
+		{
+			name:           "2 Failed Controls",
+			summaryDetails: reportsummary.SummaryDetails{},
+			controls: []resourcesresults.ResourceAssociatedControl{
+				resourcesresults.ResourceAssociatedControl{
+					ControlID: "control-1",
+					Name:      "Control 1",
+					Status:    apis.StatusInfo{},
+					ResourceAssociatedRules: []resourcesresults.ResourceAssociatedRule{
+						{
+							Name:      "Rule 1",
+							Status:    "failed",
+							SubStatus: "configuration",
+
+							Paths: []armotypes.PosturePaths{
+								{
+									FailedPath: "some-path1",
+								},
+								{
+									FailedPath: "random-path1",
+								},
+							},
+						},
+					},
+				},
+				resourcesresults.ResourceAssociatedControl{
+					ControlID: "control-2",
+					Name:      "Control 2",
+					Status:    apis.StatusInfo{},
+					ResourceAssociatedRules: []resourcesresults.ResourceAssociatedRule{
+						{
+							Name:      "Rule 2",
+							Status:    "failed",
+							SubStatus: "configuration",
+							Paths: []armotypes.PosturePaths{
+								{
+									FailedPath: "some-path2",
+								},
+								{
+									FailedPath: "random-path2",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedLen: 2,
+		},
+		{
+			name:           "One failed control",
+			summaryDetails: reportsummary.SummaryDetails{},
+			controls: []resourcesresults.ResourceAssociatedControl{
+				resourcesresults.ResourceAssociatedControl{
+					ControlID: "control-1",
+					Name:      "Control 1",
+					Status:    apis.StatusInfo{},
+					ResourceAssociatedRules: []resourcesresults.ResourceAssociatedRule{
+						{
+							Name:      "Rule 1",
+							Status:    "passed",
+							SubStatus: "configuration",
+
+							Paths: []armotypes.PosturePaths{
+								{
+									FailedPath: "some-path1",
+								},
+								{
+									FailedPath: "random-path1",
+								},
+							},
+						},
+					},
+				},
+				resourcesresults.ResourceAssociatedControl{
+					ControlID: "control-2",
+					Name:      "Control 2",
+					Status:    apis.StatusInfo{},
+					ResourceAssociatedRules: []resourcesresults.ResourceAssociatedRule{
+						{
+							Name:      "Rule 2",
+							Status:    "failed",
+							SubStatus: "configuration",
+							Paths: []armotypes.PosturePaths{
+								{
+									FailedPath: "some-path2",
+								},
+								{
+									FailedPath: "random-path2",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedLen: 1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rows := generateResourceRows(tt.controls, &tt.summaryDetails)
+			assert.Equal(t, tt.expectedLen, len(rows))
+			// assert.Equal(t, updatedPaths, testcase.expectedPaths)
```

Comment:
why is that commented out?

Reply:
I forgot to remove it, thanks for pointing this out.